src/main/scala/Column.scala

```diff
@@ -288,6 +366,708 @@ case class Column private (
+    dateTimeParser(
+      s"${xmlSchema}gMonth",
+      "invalid_gMonth",
+      value
+    )
+  }
+
+  def processGMonthDay(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gMonthDay",
+      "invalid_gMonthDat",
+      value
+    )
+  }
+
+  def processGYear(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gYear",
+      "invalid_gYear",
+      value
+    )
+  }
+  def processGYearMonth(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}gYearMonth",
+      "invalid_gYearMonth",
+      value
+    )
+  }
+
+  def processTime(
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    dateTimeParser(
+      s"${xmlSchema}time",
+      "invalid_time",
+      value
+    )
+  }
+
+  def dateTimeParser(
+      datatype: String,
+      warning: String,
+      value: String
+  ): Either[ErrorWithoutContext, ZonedDateTime] = {
+    val dateFormatObject = DateFormat(format.flatMap(f => f.pattern), datatype)
+    dateFormatObject.parse(value) match {
+      case Right(value) => Right(value)
+      case Left(error)  => Left(ErrorWithoutContext(warning, error))
+    }
+  }
+
+  def addErrorIfRequiredValueAndValueEmpty(
+      value: String,
+      rowNumber: Long
+  ): Unit = {
+    if (required && value.isEmpty) {
+      errors :+= ErrorWithCsvContext(
+        "Required",
+        "schema",
+        rowNumber.toString,
+        columnOrdinal.toString,
+        value,
+        s"required => $required"
+      )
+    }
+  }
+
+  def validate(
+      value: String,
+      rowNumber: Long
+  ): WarningsAndErrors = {
+    if (nullParam.contains(value)) {
+      addErrorIfRequiredValueAndValueEmpty(value, rowNumber)
+    } else {
+      val values = separator match {
+        case Some(separator) => value.split(separator)
+        case None            => Array[String](value)
+      }
+      val datatypeBaseField = datatype.path("base")
+      val datatypeOfColumn = if (datatypeBaseField.isMissingNode) {
+        datatype.get("@id").asText()
+      } else {
+        datatypeBaseField.asText()
+      }
+      val parserForDataType = datatypeParser(datatypeOfColumn)
+      for (v <- values) {
+        parserForDataType(v) match {
+          case Left(errorMessageContent) => {
+            errors = errors :+ ErrorWithCsvContext(
+              errorMessageContent.`type`,
+              "schema",
+              rowNumber.toString,
+              columnOrdinal.toString,
+              s"'$v' - ${errorMessageContent.content}",
+              datatype.toPrettyString
+            )
+          }
+          case Right(s) => {}
+        }
+//        val (value, warning) = //CREATE DATATYPE PARSERS
+//        val format = ""
+//        val (value, warning) = Column.DatatypeParser(s)(value, format)
+      }
+    }
+    WarningsAndErrors(Array(), Array())
```

Comment:
Do we not want to pass the errors and warnings that we've generated up?

Reply:
Yes they should be pass back up, but this function is not complete and thus codes like this. These are problems associated with breaking up a large feature to smaller PRs.  For time being I have returned the errors and warnings I have.